tests/integration/ha_tests/test_ha.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import time
+
+import pytest
+import pytest_asyncio
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.ha_tests.helpers import (
+    count_primaries,
+    deploy_and_scale_application,
+    deploy_and_scale_mongodb,
+    fetch_replica_set_members,
+    get_application_name,
+    get_mongo_client,
+    get_process_pid,
+    get_replica_set_primary,
+    get_units_hostnames,
+    mongod_ready,
+    relate_mongodb_and_application,
+    send_signal_to_pod_container_process,
+)
+from tests.integration.helpers import APP_NAME
+
+MONGODB_CONTAINER_NAME = "mongod"
+MONGOD_PROCESS_NAME = "mongod"
+MEDIAN_REELECTION_TIME = 12
+TEST_DB = "continuous_writes_database"
+TEST_COLLECTION = "test_collection"
+
+
+@pytest_asyncio.fixture
+async def continuous_writes(ops_test: OpsTest) -> None:
+    """Starts continuous writes to the MySQL cluster for a test and clear the writes at the end."""
+    application_name = await get_application_name(ops_test, "application")
+
+    application_unit = ops_test.model.applications[application_name].units[0]
+
+    clear_writes_action = await application_unit.run_action("clear-continuous-writes")
+    await clear_writes_action.wait()
+
+    start_writes_action = await application_unit.run_action("start-continuous-writes")
+    await start_writes_action.wait()
+
+    yield
+
+    clear_writes_action = await application_unit.run_action("clear-continuous-writes")
```

Comment:
Should we stop writing before clear? For example, if the test failed, maybe we should stop writes and cleanup data after the test.

Reply:
`clear-continuous-writes` action should also call `_on_stop_continuous_writes_action()`, so it should stop it on its own.